deepchem/hyper/gaussian_process.py

```diff
@@ -277,7 +284,11 @@ def optimizing_function(**placeholders):
       # Add it on to the information needed for the constructor
       hyper_parameters["model_dir"] = model_dir
       model = self.model_builder(**hyper_parameters)
-      model.fit(train_dataset)
+      try:
+        model.fit(train_dataset, nb_epoch=nb_epoch)

```

Comment:
One question that comes to mind is whether this snippet will work for models that don't have `nb_epoch` defined. Like for the random forest models. I think our other tests for hyperparam opt are still passing so we're probably all right but might be good to double check.

Reply:
Locally, the tests for hyperparam opt pass. To be sure, I adapted the random forest model used in `test_gaussian_hyperparam_opt.py` and `test_grid_hyperparam_opt.py` to a stand alone script, and tested behavior with and without specifying `nb_epoch`. The optimization ran in both cases, and returned the correct `best_hyperparams` and `best_model` (i.e., the one with the highest pearson_r2_score). Setting `nb_epoch` to some absurdly high value did not affect execution time, so it is being ignored during optimization.

If there is a standard dataset to test optimization with, let me know and I'll drop that dataset in to test. I used the same randomly generated Numpy arrays, so not a good test of actual optimization.

Reply:
Ah great, thank you for the extra checks! That should work for this PR.

We don't currently have a standard dataset to test optimization with right now. It's a little tricky since the smallest molnet datasets are ~1K compounds and would be a little too slow for the test suite. Perhaps we should use something like Iris or tiny dataset as a standard benchmark though. Might be a good avenue for future improvements